rfcs/0000-graphql-api.md

```diff
@@ -0,0 +1,431 @@
+## Summary
+[summary]: #summary
+
+A GraphQL api design to allow the wallet app (and future similar applications) to communicate with a full node.
+
+## Motivation
+
+[motivation]: #motivation
+
+Currently, the main use case for this is to support the development of a wallet application which can
+communicate with a full node which is running either remotely or locally in another process.
+
+Related Goals:
+- Make app development using the api relatively easy and fast.
+- Make it easier for people to communicate programmatically with a full node.
+- Consumers of the api should not be required to write in a specific language.
+- Eventually be able to rewrite the command-line client to use the same api.
+
+## Detailed design
+
+[detailed-design]: #detailed-design
+
+This design is heavily influenced by which data is stored on the node and which is intended to be stored by the client.
+Most importantly, we assume the node knows about some private keys which it associates to public keys for which it will
+try to store related transactions indefinitely (where "related" means being sent to or from the public key). We may also
+need to store information about blocks that these keys have created.
+We will likely want to at least optionally cache the history of the blocks/payments on the client side as well,
+just to avoid needing to query it all every time.
+
+Note: There are several custom scalars defined at the top for readability. These will be `String`s in the final implementation due to the complications involved in the encoding of custom scalars not being expressed in the schema and needing to be implemented symmetrically on the client and server.
+
+```graphql
+scalar Date
```

Comment:
a string that contains a number for the number of milliseconds since "epoch" 1970